components/brave_ads/core/internal/diagnostics/diagnostic_manager.cc

```diff
@@ -30,6 +31,7 @@ DiagnosticManager::DiagnosticManager() {
   SetEntry(std::make_unique<OptedInToBraveNewsAdsDiagnosticEntry>());
   SetEntry(std::make_unique<OptedInToNewTabPageAdsDiagnosticEntry>());
   SetEntry(std::make_unique<OptedInToNotificationAdsDiagnosticEntry>());
+  SetEntry(std::make_unique<OptedInToSearchAdsDiagnosticEntry>());
```

Comment:
nit: for consistency with rest of `brave_ads/core` could we please use `OptedInToSearchResultAdsDiagnosticEntry`. Same for any other usage.